pkg/util/controllerfinder/pods_finder.go

```diff
@@ -48,37 +48,38 @@ func (r *ControllerFinder) GetPodsForRef(apiVersion, kind, ns, name string, acti
 		}
 		workloadReplicas = *rs.Spec.Replicas
 		workloadUIDs = append(workloadUIDs, rs.UID)
-	// Deployment, get the corresponding ReplicaSet UID
-	case ControllerKindDep.Kind:
-		rss, err := r.getReplicaSetsForDeployment(apiVersion, kind, ns, name)
+	// statefulset, rc, cloneSet
+	case ControllerKindSS.Kind, ControllerKindRC.Kind, ControllerKruiseKindCS.Kind, ControllerKruiseKindSS.Kind:
+		obj, err := r.GetScaleAndSelectorForRef(apiVersion, kind, ns, name, "")
 		if err != nil {
 			return nil, -1, err
-		}
-		if len(rss) == 0 {
+		} else if obj == nil {
 			return nil, 0, nil
 		}
+		workloadReplicas = obj.Scale
+		workloadUIDs = append(workloadUIDs, obj.UID)
+	// Deployment and others crds
```

Comment:
there is special logic to extract replicaset for deployment, is it better to separate deployment handling from general scale resource handling?